nemo_text_processing/text_normalization/de/taggers/cardinal.py

```diff
@@ -0,0 +1,207 @@
+
+        def hundred_non_zero():
+            return (graph_digit_no_one + insert_space | pynini.cross("1", "ein ")) + pynutil.insert("hundert") + (
+                pynini.closure(insert_space + pynutil.insert(AND), 0, 1) + insert_space + tens_no_zero()
+                | pynutil.delete("00")
+            ) | pynutil.delete("0") + tens_no_zero()
+
+        def thousand():
+            return (hundred_non_zero() + insert_space + pynutil.insert("tausend") | pynutil.delete("000")) + (
+                insert_space + hundred_non_zero() | pynutil.delete("000")
+            )
+
+        graph_million = pynini.union(
+            hundred_non_zero() + insert_space + pynutil.insert("million") + pynini.closure(pynutil.insert("en"), 0, 1),
+            pynutil.delete("000"),
+        )
+
+        graph_billion = pynini.union(
+            hundred_non_zero()
+            + insert_space
+            + pynutil.insert("milliarde")
+            + pynini.closure(pynutil.insert("n"), 0, 1),
+            pynutil.delete("000"),
+        )
+
+        graph_trillion = pynini.union(
+            hundred_non_zero() + insert_space + pynutil.insert("billion") + pynini.closure(pynutil.insert("en"), 0, 1),
+            pynutil.delete("000"),
+        )
+
+        graph_quadrillion = pynini.union(
+            hundred_non_zero()
+            + insert_space
+            + pynutil.insert("billiarde")
+            + pynini.closure(pynutil.insert("n"), 0, 1),
+            pynutil.delete("000"),
+        )
+
+        graph_quintillion = pynini.union(
+            hundred_non_zero()
+            + insert_space
+            + pynutil.insert("trillion")
+            + pynini.closure(pynutil.insert("en"), 0, 1),
+            pynutil.delete("000"),
+        )
+
+        graph_sextillion = pynini.union(
+            hundred_non_zero()
+            + insert_space
+            + pynutil.insert("trilliarde")
+            + pynini.closure(pynutil.insert("n"), 0, 1),
+            pynutil.delete("000"),
+        )
+        graph = pynini.union(
+            graph_sextillion
+            + insert_space
+            + graph_quintillion
+            + insert_space
+            + graph_quadrillion
+            + insert_space
+            + graph_trillion
+            + insert_space
+            + graph_billion
+            + insert_space
+            + graph_million
+            + insert_space
+            + thousand()
+        )
+
+        self.graph = (
+            ((NEMO_DIGIT - "0" + pynini.closure(NEMO_DIGIT, 0)) - "0" - "1")
+            @ pynini.cdrewrite(pynini.closure(pynutil.insert("0")), "[BOS]", "", NEMO_SIGMA)
+            @ NEMO_DIGIT ** 24
+            @ graph
+            @ pynini.cdrewrite(delete_space, "[BOS]", "", NEMO_SIGMA)
+            @ pynini.cdrewrite(delete_space, "", "[EOS]", NEMO_SIGMA)
+        )
+        self.graph |= graph_zero | pynini.cross("1", "eins")
+
+        # self.graph = pynini.cdrewrite(pynutil.delete(separator), "", "", NEMO_SIGMA) @ self.graph
```

Comment:
remove if not needed

Reply:
I would like to keep it since I might want to need to use separators in future